File: src/main/java/net/hectus/neobb/turn/legacy_game/block/LTPurpleWool.java
package net.hectus.neobb.turn.legacy_game.block;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.NeutralClazz;
import net.hectus.neobb.turn.default_game.attributes.function.AttackFunction;
import net.hectus.neobb.turn.default_game.block.BlockTurn;
import org.bukkit.block.Block;

public class LTPurpleWool extends BlockTurn implements AttackFunction, NeutralClazz {
    public LTPurpleWool(NeoPlayer player) { super(player); }
    public LTPurpleWool(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 15;
    }

    @Override
    public boolean unusable() {
        return isDummy() || player.game.history().size() < 10;
    }

    @Override
    public void apply() {
        player.game.win(player);
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/legacy_game/block/LTPurpleWool.kt
package net.hectus.neobb.modes.turn.legacy_game.block

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.NeutralClazz
import net.hectus.neobb.modes.turn.default_game.attribute.function.AttackFunction
import net.hectus.neobb.modes.turn.default_game.block.BlockTurn
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.block.Block

class LTPurpleWool(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), AttackFunction, NeutralClazz {
    override val cost: Int = 15

    override fun unusable(): Boolean {
        return isDummy() || player!!.game.history.size < 10
    }

    override fun apply() {
        player!!.game.win(player)
    }
}
